README.md

```diff
@@ -1,2 +1,203 @@
+
+- ROOT's package manager prototype
+
+root-get is the project centric package manager for ROOT. The application provides simplified build and install process for ROOT packages and modules. 
+
+## Getting started
+
+## Manifest file
+The installation procedure in root-get has manifest files at its base. Manifest file is structured as yml file and provides the necessary attributes regarding an entity. (hereafter entity refers to package or module)
+Package manifest contains the package attributes along with specifications regarding its modules.  
+Module manifest contains attributes for a specific module.
+The necessary attributes in any manifest file are :
+- name (module/package)
+- packageurl (In case of downloading the entity and using it)
+- path (The absolute path for the entity if present in system)
+- deps (Dependencies for the entity)
+- publicheaders (Headers for the entity)
+- sources 
+- targets 
+
+## Environment variables
+Setting environment variables before using root-get is important to allow root-get to search for various required files from the system.
+Important environment variables :
+- ROOTSOURCES
+
+The rootbase source files directory. Useful in build and install of root packages and modules.
+
+- ROOT_PKG_PATH
+
+The directory path where user can get the installed package/module.
+
+- ROOTSYS
+
+The rootbase build files directory
+
+- ROOT_MODULES
+
+The directory with all the module manifest (.yml) files
+
+Currently the installation procedure makes use of [rootbase](https://github.com/root-project/root) source files for root packages and modules.
+- rootbase package install procedure
+
+The routine uses package specific manifest file and installs each module from the manifest.
+- rootbase module install procedure
+
+The routine requires the module specific manifest file and proceeds with it.
+
+root-get extensively makes use of cache to avoid re-building of entities. The already built packages are stored in cache directory.
+
+## External entity handling routine
+
+- External entity - Entity outside rootbase scope.
+
+The entity can be present in user system or may not be present on the system. For external entity also, root-get requires manifest file.
+The routine checks for entity presence in user system otherwise uses downloader routine to download the entity repository from GitHub(currently supported) and returns the path for the new downloaded directory. The further process takes the similar path as per internal entity routine.
+
+User can provide own manifest file with all neccessary attributes present in it to plug an external entity. To create your own manifest file please follow the instructions below
+
+- Creating module manifest file :
+```
+module:
+  name: 'modulename'
+  packageurl: 'link/to/github/repository'
+  tag: 0.0.0 
+  path: 'path/to/module/in/system'
+  publicheaders: 'if present'
+  sources: 'if present'
+  targets: 'modulename'
+  deps: 'dependencies'
+```
+
+- Example module.yml :
```

Comment:
Lets use here libhello?

Reply:
yes, actually it would be good thing to do.! thanks.!